<a href="https://colab.research.google.com/github/NguyenTanTan/Bai-tap-python/blob/python/Ki%E1%BB%83m_%C4%91%E1%BB%8Bnh_gi%E1%BA%A3_thuy%E1%BA%BFt_tr%C3%AAn_m%E1%BA%ABu_%C4%91%C6%A1n_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Lý thuyết**:
* **t-test** 
* Công thức:
   * $t=\frac{\bar x-\mu}{\frac{S_x}{\sqrt{n}}}$
   * t: độ lệch của trung bình mẫu với trung bình của tập hợp
   * $\bar x$:  trung bình của mẫu (giá trị muốn kiểm định)
   * $\mu$: trung bình của tập hợp
   * $S_x$: độ lệch chuẩn
   * n: kích thước mẫu
* Dùng phương pháp ***p-value***; so sánh ***p-value*** với $α$ (mức tin cậy)
* Các bước:
  * Tạo giả thuyết
  * Tính toán **p-value**:
   * Dùng hàm: ***scipy.stats.ttest_1samp***
  * So sánh với $α$:
   * **p-value < α**  bác bỏ giả thuyết không
   * **p-value > α**  không bác bỏ giả thuyết không

  * Kết luận


* Xây dựng kiểm định giả thuyết trên mẫu đơn của trung bình:

TH1: One-tailed:  $\mu<\mu_0$ or $\mu>\mu_0$

TH2: Two-tailed: $\mu\neq\mu_0$

**TH1: One-tailed**


**Tạo giả thuyết:**
* Cho tập dữ liệu mẫu là chiều dài các đoạn dây
* Cho giả thuyết không là 50. Giả thuyết nghịch là bé hơn 50. Và mức ý nghĩa là 0.1, mẫu là 20:
 * $H_0=50$
 * $H_a<50$
 * $α=0,1$
 * n=20

In [ ]:
import numpy as np
from scipy import stats as st
import random

In [ ]:
# Tạo tập dữ liệu
data=np.random.randint(45,55,10)
print(data)

[50 50 51 54 54 54 54 53 46 46]


**Dùng hàm st.ttest_1samp **
* Trả về kết quả gồm: 
1. statistic: giá trị sai số của trung bình mẫu  với trung bình của tập hợp
2. pvalue : dùng so sánh với $α$

In [ ]:
# Tính p-value
st.ttest_1samp(data, popmean=50) 

Ttest_1sampResult(statistic=1.1894734808191643, pvalue=0.2646862893386352)

* Kết luận: 
 Giá trị **p-value** = 0.26 > $α$=0.1 nên không thể bác bỏ giả thuyết. 

**TH2: Two-tailed**

* Cho tập dữ liệu mẫu là chiều dài các đoạn dây
* Cho giả thuyết không là 50. Giả thuyết nghịch là bé hơn 50. Và mức ý nghĩa là 0.1, mẫu là 20:
 * $H_0=50$
 * $H_a\neq50$
 * $α=0,1$
 * n=20